plugins/amazonq/chat/jetbrains-community/src/software/aws/toolkits/jetbrains/services/cwc/editor/context/project/ProjectContextController.kt

```diff
@@ -0,0 +1,58 @@
+// Copyright 2024 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package software.aws.toolkits.jetbrains.services.cwc.editor.context.project
+
+import com.intellij.openapi.Disposable
+import com.intellij.openapi.components.Service
+import com.intellij.openapi.components.service
+import com.intellij.openapi.project.Project
+import com.intellij.openapi.util.Disposer
+import kotlinx.coroutines.launch
+import software.aws.toolkits.core.utils.getLogger
+import software.aws.toolkits.core.utils.warn
+import software.aws.toolkits.jetbrains.core.coroutines.disposableCoroutineScope
+import software.aws.toolkits.jetbrains.services.codewhisperer.settings.CodeWhispererSettings
+
+@Service(Service.Level.PROJECT)
+class ProjectContextController(val project: Project) : Disposable {
+    private val encoderServer: EncoderServer = EncoderServer(project)
+    private val projectContextProvider: ProjectContextProvider = ProjectContextProvider(project, encoderServer)
+    private val scope = disposableCoroutineScope(this)
```

Comment:
inject the coroutinescope in the service constructor instead of using this deprecated function

Reply:
@Will-ShaoHua also mentioned above, will followup on this